### PR TITLE
HAAR-834: Move get all groups to external users (from auth)

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsController.kt [hunk 1]
```diff
@@ -35,6 +35,32 @@ class GroupsController(
   private val groupsService: GroupsService
 ) {
 
+  @GetMapping("/groups")
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_OAUTH_USERS', 'ROLE_AUTH_GROUP_MANAGER')")
+  @Operation(
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsController.kt:40]: There was no role in auth, but I have added for consistency

File: src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsController.kt [hunk 2]
```diff
@@ -73,10 +99,7 @@ class GroupsController(
     @Parameter(description = "The group code of the group.", required = true)
     @PathVariable
     group: String
-  ): GroupDetails {
-    val returnedGroup: Group = groupsService.getGroupDetail(group)
-    return GroupDetails(returnedGroup)
-  }
+  ): GroupDetails = GroupDetails(groupsService.getGroupDetail(group))
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsController.kt:102]: tidy up

File: src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsControllerIntTest.kt [hunk 3]
```diff
@@ -8,12 +8,57 @@ import org.junit.jupiter.api.Test
...
+    }
+
+    @Test
+    fun `All Groups endpoint returns all possible groups`() {
+      webTestClient
+        .get().uri("/groups")
+        .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_OAUTH_USERS")))
+        .exchange()
+        .expectStatus().isOk
+        .expectHeader().contentType(MediaType.APPLICATION_JSON)
```
Comment (ravmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsControllerIntTest.kt:53]: static import

**Review by ravmoj — DISMISSED** [review]
lgtm

**Pull request merged**